src/clear_display.c

```diff
@@ -48,20 +48,68 @@
 #include "tlf_curses.h"
 #include "ui_utils.h"
 
-static char terminal1[88] = "";
-static char terminal2[88] = "";
-static char terminal3[88] = "";
-static char terminal4[88] = "";
+void clear_display(void);
 
 
-void init_terminal_strings(void) {
-    strcat(terminal1, backgrnd_str);
-    strcat(terminal2, backgrnd_str);
-    strcat(terminal3, backgrnd_str);
-    strcat(terminal4, backgrnd_str);
+void clear_line(int row) {
+    mvaddstr(row, 0, backgrnd_str);
+}
+
+static char *terminal1;
+static char *terminal2;
+static char *terminal3;
+static char *terminal4;
+
+void init_keyer_terminal(void) {
+    terminal1 = g_strdup("");
+    terminal2 = g_strdup("");
+    terminal3 = g_strdup("");
+    terminal4 = g_strdup("");
+}
+
+static void show_keyer_terminal() {
+    attron(modify_attr(COLOR_PAIR(C_LOG) | A_STANDOUT));
+    for (int i = 1; i < 6; i++)
+	clear_line(i);
+
+    mvaddstr(1, 0, terminal1);
+    mvaddstr(2, 0, terminal2);
+    mvaddstr(3, 0, terminal3);
+    mvaddstr(4, 0, terminal4);
+}
+
+/*
+ * add 'buffer' to the keyer terminal while scrolling one line up
+ * and refreshing the display
+ */
+static void add_line_to_keyer_terminal(char *buffer) {
+
+    if (terminal1)
```

Comment:
just a minor note: g_free does the NULL check internally.

Reply:
Thanks for the reminder.